src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -565,24 +623,134 @@ private void InterpretDup()
 
         private void InterpretLoadLocal(int index)
         {
+            Debug.Assert(index >= 0);
             _stack.Push(_locals[index]);
         }
 
         private void InterpretStoreLocal(int index)
         {
+            Debug.Assert(index >= 0);
             _locals[index] = PopWithValidation();
         }
 
+        private void InterpretLoadArgument(int index)
+        {
+            Debug.Assert(index >= 0);
+
+            TypeDesc argument = default(TypeDesc);
+
+            if (!_method.Signature.IsStatic)
+            {
+                if (index == 0)
+                    argument = _method.OwningType;
+                else
+                    argument = _method.Signature[index - 1];
+            }
+            else
+            {
+                argument = _method.Signature[index];
+            }
+
+            switch (argument.Category)
+            {
+                case TypeFlags.Boolean:
+                case TypeFlags.Char:
+                case TypeFlags.SByte:
+                case TypeFlags.Byte:
+                case TypeFlags.Int16:
+                case TypeFlags.UInt16:
+                case TypeFlags.Int32:
+                case TypeFlags.UInt32:
+                    _stack.Push(StackItem.FromInt32(GetArgument<int>(index)));
+                    break;
+                case TypeFlags.Int64:
+                case TypeFlags.UInt64:
+                    _stack.Push(StackItem.FromInt64(GetArgument<long>(index)));
+                    break;
+                case TypeFlags.IntPtr:
+                case TypeFlags.UIntPtr:
+                    _stack.Push(StackItem.FromNativeInt(GetArgument<IntPtr>(index)));
+                    break;
+                case TypeFlags.Single:
+                case TypeFlags.Double:
+                    _stack.Push(StackItem.FromDouble(GetArgument<double>(index)));
+                    break;
+                case TypeFlags.ValueType:
+                case TypeFlags.Nullable:
+                    _stack.Push(StackItem.FromValueType(GetArgument<ValueType>(index)));
+                    break;
+                case TypeFlags.Enum:
+                    {
+                        StackValueKind kind = GetStackValueKind(argument.UnderlyingType);
+                        switch (kind)
+                        {
+                            case StackValueKind.Int32:
+                                _stack.Push(StackItem.FromInt32(GetArgument<int>(index)));
+                                break;
+                            case StackValueKind.Int64:
+                                _stack.Push(StackItem.FromInt64(GetArgument<long>(index)));
+                                break;
+                            default:
```

Comment:
Enums in IL can be NativeInt, Float or Double too. It would be better to call this switch recursively (or to do `goto` to before the switch) to not miss any of the cases.

Reply:
`GetStackValueKind` duplicates the logic from this method. My point was to avoid this duplication.